### PR TITLE
{tools}[foss/2021b,foss/2022a] Horovod w/ TensorFlow

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

Add Horovod for all TensorFlow versions where we don't have one yet.

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
taurusi8015 - Linux CentOS Linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), Python 2.7.5
See https://gist.github.com/Flamefire/7110e144feae9430212973827e80c933 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
taurusml4 - Linux RHEL 7.6, POWER, 8335-GTX (power9le), 6 x NVIDIA Tesla V100-SXM2-32GB, 440.64.00, Python 2.7.5
See https://gist.github.com/Flamefire/26281489c4554103f47b2718d80632cf for a full test report.

File: easybuild/easyconfigs/h/Horovod/Horovod-0.22.1-foss-2021a-CUDA-11.3.1-TensorFlow-2.5.3.eb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+]
+
+use_pip = True
+sanity_pip_check = True
+
+preinstallopts = 'HOROVOD_WITH_MPI=1 HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_GPU_BROADCAST=NCCL '
+preinstallopts += 'HOROVOD_WITH_TENSORFLOW=1 HOROVOD_WITHOUT_PYTORCH=1 HOROVOD_WITHOUT_MXNET=1 '
+
+# Bug in CMake causes a race condition on horovod_cuda_kernels_generated_cuda_kernels.cu.o.NVCC-depend
+maxparallel = 1
```
[the remark this answers is not in the document]
Comment (Flamefire) [reply easybuild/easyconfigs/h/Horovod/Horovod-0.22.1-foss-2021a-CUDA-11.3.1-TensorFlow-2.5.3.eb:32]: Only for Horovod 0.24+, see https://github.com/horovod/horovod/pull/3261 mentioned in #18262   
It's actually a bug in how Horovod used something of CMake which is outdated and hence won't be fixed.

[the remark this answers is not in the document]
Comment (jfgrimm) [reply easybuild/easyconfigs/h/Horovod/Horovod-0.22.1-foss-2021a-CUDA-11.3.1-TensorFlow-2.5.3.eb:32]: :+1:

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
bear-pg0203u29a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), 1 x NVIDIA NVIDIA A100-SXM4-80GB, 520.61.05, Python 3.6.8
See https://gist.github.com/branfosj/1a59d216a6c6c3677cdbfd0ea731adb0 for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @Flamefire!

**Pull request merged**